File: problems/0007-Reverse-integer/javascript/solution.js
var reverse = function (x) {
  let ans = 0;
  while (x != 0) {
    ans = ans * 10 + (x % 10);
    x = Math.abs(x) < 10 ? 0 : Math.trunc(x / 10);
  }
  if (ans > Math.pow(2, 31) - 1 || ans < -Math.pow(2, 31)) {
    // checking if the no. lie within bounds
    return 0;
  }
  return ans;
};

File: problems/0007-Reverse-integer/C++/solution.cpp

// most optimized solution
class Solution
{
public:
    int reverse(int x)
    {
        long rev_num = 0;
        while (x)
        {
            rev_num = rev_num * 10 + x % 10;
            x = x / 10;
        }
        if (rev_num > INT_MAX || rev_num < INT_MIN) // checking bounds of the number and returning 0 in case
            return 0;
        return rev_num;
    }
};